boards/arm/arty/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+# SPDX-License-Identifier: Apache-2.0
+
+if((CONFIG_BOARD_ARTY_A7_ARM_DESIGNSTART_M1) AND (CONFIG_BUILD_OUTPUT_BIN))
+  # Generate zephyr.mem verilog memory hex dump file for initialising ITCM in
+  # Xilinx Vivado.
+  #
+  # This ought to be done using the objcopy verilog bfd, but it contains a bug
+  # affecting endianness: https://sourceware.org/bugzilla/show_bug.cgi?id=25202
+  #
+  # Instead we use bin2hex from the SiFive elf2hex package, if available.
+  # https://github.com/sifive/elf2hex
+  find_program(BIN2HEX ${CROSS_COMPILE_TARGET}-bin2hex)
+
+  if(NOT ${BIN2HEX} STREQUAL BIN2HEX-NOTFOUND)
+    set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+      COMMAND ${BIN2HEX}
+      ARGS --bit-width 32
+      ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}.bin
+      ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}.mem
+      WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+      )
+  endif()
```

Comment:
just wondering.
a user not having `bin2hex` installed, and have `CONFIG_BUILD_OUTPUT_BIN=y` might wonder why no hex file is being produced.

Could you add a message that `hex-file` output can not be produced. (build should not fail imho).
Either at the CMake configure stage or as an extra post build command.

One thing that favors build time, is that user will be notified on each `ninja/make` invocation that a hex file cannot be produced.
Also if user uses `west build` then a CMake message might not be noticed, whereas a build time message is more likely to get attention.

As example:
```
bin2hex could not be found, hex file cannot be produce.
Please install bin2hex or disable hex-file output with `CONFIG_BUILD_OUTPUT_BIN=n`
```

Reply:
@tejlmand Thank you for your feedback. I have added a couple of messages to the board `CMakeLists.txt` file. I opted to output these during configuration, since most people will just use the .elf file for development (and the .mem file for deployment).

Reply:
thanks.